### PR TITLE
Update documentation: add supported entities section

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Update documentation: add supported entities section
  - Disable inherited members for quicker docs generation
  - Fix rough docstring cases
  - Add generated autosummary to .gitignore
  - Reformat pre-commit config
  - Extend make clean with documentation cleanup command

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by KJhellico** [comment]
It looks like we need to fix all links in comments containing Unicode characters, for Sphinx to process them correctly. Currently, links like https://zh.wikipedia.org/wiki/中华人民共和国节日与公众假期 are broken.

**Comment by arkid15r** [comment]
> It looks like we need to fix all links in comments containing Unicode characters, for Sphinx to process them correctly. Currently, links like [https://zh.wikipedia.org/wiki/中华人民共和国节日与公众假期](https://zh.wikipedia.org/wiki/%E4%B8%AD%E5%8D%8E%E4%BA%BA%E6%B0%91%E5%85%B1%E5%92%8C%E5%9B%BD%E8%8A%82%E6%97%A5%E4%B8%8E%E5%85%AC%E4%BC%97%E5%81%87%E6%9C%9F) are broken.

Yes, we have quite a few entities with malformed documentation. This is the first step towards fixing that.

File: holidays/countries/thailand.py [hunk 1]
```diff
@@ -31,104 +31,101 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
...
+        - [Asarnha Bucha]
+            https://www.onab.go.th/th/content/category/detail/id/73/iid/3397
+        - [Buddhist Lent Day]
+            https://www.onab.go.th/th/content/category/detail/id/73/iid/3395
+        - [Royal Ploughing Ceremony]
+            https://en.wikipedia.org/wiki/Royal_Ploughing_Ceremony
+            https://www.lib.ru.ac.th/journal/may/may_phauchmongkol.html
+            https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2540.aspx
+        - [Royal Thai Armed Forces Day]
+            https://th.wikipedia.org/wiki/วันกองทัพไทย
```
Comment (KJhellico) [inline holidays/countries/thailand.py:95]:
```suggestion
            `<https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2540.aspx>`_
        - [Royal Thai Armed Forces Day]
            `<https://th.wikipedia.org/wiki/วันกองทัพไทย>`_
```

**Review by arkid15r — COMMENTED** [review]
I've merged your suggestions 👍 

This PR goal is introducing the supported entities docs. We need to make sure RTD builds it successfully. Fixing actual documentation is the next step. I'd also like to reconsider current format for args/returns and links.

**Review by KJhellico — APPROVED** [review]
Then let's try! LGTM 🚀

File: holidays/countries/kazakhstan.py [hunk 2]
```diff
@@ -16,10 +16,11 @@
...
     """
-    1. https://www.officeholidays.com/countries/kazakhstan/2020
-    2. https://egov.kz/cms/en/articles/holidays-calend
-    3. https://en.wikipedia.org/wiki/Public_holidays_in_Kazakhstan
-    4. https://adilet.zan.kz/rus/docs/Z010000267_/history
+    References:
+        - https://www.officeholidays.com/countries/kazakhstan/2020
+        - https://egov.kz/cms/en/articles/holidays-calend
+        - https://en.wikipedia.org/wiki/Public_holidays_in_Kazakhstan
+        - https://adilet.zan.kz/rus/docs/Z010000267\_/history  # noqa W605
```
Comment (phillipuniverse) [inline holidays/countries/kazakhstan.py:23]: Hm @arkid15r @KJhellico looks like this ignore caused a problem with Python 3.11.4 https://github.com/vacanza/python-holidays/issues/1612

**Pull request merged**